### PR TITLE
Limit the window size for desktop app, setup jest test framework

### PR DESCRIPTION
**Pull request by ashu8912**
## Testing done
still needs to be tested on larger screens

### CODE REVIEW

File: app/electron/main.ts [hunk 1]
```diff
@@ -258,6 +258,24 @@ function setMenu(i18n: I18n) {
   Menu.setApplicationMenu(menu);
 }
 
+function calculateWindowSize() {
+  const { width, height } = screen.getPrimaryDisplay().workAreaSize;
+  const [maxWidth, maxHeight] = [1920, 1080];
+  const pixelRatio = screen.getPrimaryDisplay().scaleFactor;
```
Comment (illume) [inline app/electron/main.ts:264]: scaleFactor is a Number which can be a float. So a 1.5x screen should be returned here as `1.5`.

So the case statement below should be changed into a series of ranges.

**Comment by illume** [comment]
Started a table so we can set some expectations of what good sizes would be.

| Size         | Scale     | Window size | Name        |
|--------------|-----------|-------------|-------------|
| 1366×768     | 100%      |             | Popular     |
| 1920x1080    | 100%      |             | HD          |
| 1920x1080    | 200%      |             | 4K zoomed   |
| 2880x1800    | 100%      |             | Mac retina  |
| 1440x900     | 200%      |             | Mac retina  |
| 3240x2160    | 200%      |             | SB3         |
| 3240x2160    | 100%      |             | SB3 0 scale |
| 3840x2160    | 100%      |             | 4K          |

**Comment by illume** [comment]
> for displays larger then 1920x1080 we should limit the windows size to 1920x1080

Limiting it to 1920x1080 on large screens sounds ok to me. I wonder if anyone uses Headlamp as a dashboard? For a dashboard mode, full screen would be important. Can't think of another reason on large screens why someone would prefer a giant window with Headlamp as it is.

**Comment by illume** [comment]
I made a test using a table of expected results. But haven't checked if those results are correct.

One change I also made I wanted to ask you about... The height should probably not be limited. Instead it should probably use the full height of the work area, because some views could be quite long.  What do you think?

**Comment by illume** [comment]
I put some margin around the height and width... mainly because on WSL it wasn't fitting on screen. But maybe other systems have toolbars/docks. It kind of looks a bit squished height wise on my Mac when the dock is hiding.

Also put the max height back, because on large screens it's kind of silly.

File: app/electron/main.ts [hunk 2]
```diff
@@ -342,7 +338,11 @@ function startElecron() {
         protocol: 'file:',
         slashes: true,
       });
-    const { width, height } = calculateWindowSize();
+
+    // WSL has a problem with full size window placement, so make it smaller.
+    const windowMargin = isWSL();
+    const { width, height } = windowSize(screen.getPrimaryDisplay().workAreaSize, windowMargin);
```
Comment (joaquimrocha) [inline app/electron/main.ts:344]: Nit: Reading this, I'd assume windowMargin is a numeric value. It's good practice for booleans to be indicated as such by  using `is` or `with` or similar prefixes.

File: app/package.json [hunk 3]
```diff
@@ -14,7 +14,8 @@
     "package": "npm run copy-icons && npm run compile-electron && electron-builder build --publish never",
     "dev": "npm run compile-electron && cross-env ELECTRON_DEV=1 electron .",
     "dev-only-app": "npm run compile-electron && cross-env ELECTRON_DEV=1 ELECTRON_START_URL=http://localhost:3000 EXTERNAL_SERVER=true electron .",
-    "i18n": "npx --no-install i18next ./electron/main.ts -c ./electron/i18next-parser.config.js"
+    "i18n": "npx --no-install i18next ./electron/main.ts -c ./electron/i18next-parser.config.js",
+    "test": "jest"
```
Comment (joaquimrocha) [inline app/package.json:18]: Did you forget to include jest in package.json? I see it only in the lock file.

Comment (illume) [reply app/package.json:18]: My clumsy clumsy fingers! Thanks.

File: app/electron/main.ts (hunk 2 above)
Comment (illume) [reply app/electron/main.ts]: Nit squashed withThanks.

**Review by joaquimrocha — APPROVED** [review]

**Pull request merged**